### ISSUE TITLE
Unable to set logging levels via environment variables

### ISSUE THREAD

**Issue by shahmirn** [micronaut-projects/micronaut-core#3939]
After creating a brand new project with the management feature, we noticed that we're not able to set logging levels via environment variables, and only via system properties, even though https://docs.micronaut.io/latest/guide/index.html#config says:

In addition, just like Spring and Grails, Micronaut allows overriding any property via system properties or environment variables.

and https://docs.micronaut.io/latest/guide/index.html#propertySource says:

Controlling Log Levels with Properties
Log levels can be configured via properties defined in application.yml (and environment variables)

### Task List

- [X] Steps to reproduce provided
- [ ] Stacktrace (if present) provided
- [X] Example that reproduces the problem uploaded to Github
- [X] Full description of the issue provided (see below)

### Steps to Reproduce

1. Go to https://micronaut.io/ and click generate project.
2. Select Java 8, Kotlin, and Kotlintest
3. Click on Features, and select management
4. Click on Generate Project
5. Modify application.yml and explicitly turn on the loggers endpoint, and configure the log level for foo:

```
endpoints:
    loggers:
        enabled: true
logger:
    levels:
        foo: ERROR
```
6. Assemble the jar file via:

```
./gradlew assemble
```

7. Run the project, setting an environment variable for the MICRONAUT_ENVIRONMENTS (to demo that other environment variables work), an environment variable for a log level, and a system property for another log level:

```
MICRONAUT_ENVIRONMENTS=log_bug LOGGER_LEVELS_IO=ERROR java -Dlogger.levels.bar=ERROR -jar build/libs/demo-0.1-all.jar
```

8. Confirm that you see in the startup logs:

```
Established active environments: [log_bug]
```

9. Go to http://localhost:8080/loggers 

### Expected Behaviour

All three log levels should be configured, and the /loggers endpoint should display:

```
"bar": {
"configuredLevel": "ERROR",
"effectiveLevel": "ERROR"
},
"foo": {
"configuredLevel": "ERROR",
"effectiveLevel": "ERROR"
},
"io": {
"configuredLevel": "ERROR",
"effectiveLevel": "ERROR"
}
```

### Actual Behaviour

Only the log level for foo (from application.yml) and bar (from system property) is configured, while the log level for IO is not

```
"bar": {
"configuredLevel": "ERROR",
"effectiveLevel": "ERROR"
},
"foo": {
"configuredLevel": "ERROR",
"effectiveLevel": "ERROR"
},
"io": {
"configuredLevel": "NOT_SPECIFIED",
"effectiveLevel": "INFO"
}
```

### Environment Information

- **Operating System**: MacOS Big Sur
- **Micronaut Version:** 2.0.1
- **JDK Version:**
openjdk version "1.8.0_265"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_265-b01)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.265-b01, mixed mode)

### Example Application

https://github.com/shahmirn/micronaut-bugs

**Comment by LtTempletonPeck** [comment]
I have the same problem with an AWS lambda function but adding the the log config to the `bootstrap.yml` works

**Comment by David-16** [comment]
Is this same as https://github.com/micronaut-projects/micronaut-core/issues/4580 ?

**Comment by shahmirn** [comment]
Seems to be the case? Sounds very similar.

**Comment by timyates** [comment]
Thanks for the report!

Just tested it, and this works as expected in Micronaut 3.2.4 and the latest 2.2.X branch (where a fix was applied via #4580)

Closing this, please feel free to open a new issue if you see otherwise 👍

**Comment by timyates** [comment]
There's this also, which I think is the final fix which went in to Micronaut 3.0.x

https://github.com/micronaut-projects/micronaut-core/pull/5921
